### PR TITLE
Implement handling of EphemeralAddresses 

### PR DESCRIPTION
**Pull request by ec2**

### CODE REVIEW

File: zcash_client_memory/src/error.rs [hunk 1]
```diff
@@ -65,10 +66,19 @@ pub enum Error {
...
     ReachedGapLimit(AccountId, u32),
+    #[cfg(feature = "transparent-inputs")]
+    #[error("Transparent derivation: {0}")]
+    TransparentDerivation(bip32::Error),
     #[error("Address Conversion error: {0}")]
     ConversionError(ConversionError<&'static str>),
 }
+#[cfg(feature = "transparent-inputs")]
 
+impl From<bip32::Error> for Error {
```
Comment (willemolding) [inline zcash_client_memory/src/error.rs:77]: Could this be reimplemented using the `#[from]` attribute macro? e.g. 
```
#[error("Transparent derivation: {0}")]
    TransparentDerivation(#[from] bip32::Error)
```
or does bip32::Error fail to implement some traits this requires

**Review by willemolding — COMMENTED** [review]
I'm feeling too dumb to be able to review this right now :( We might have to have a call so I can get up to speed on how some parts work

File: zcash_client_memory/src/error.rs (hunk 1 above)
Comment (ec2) [reply zcash_client_memory/src/error.rs:77]: Yeah, it doesnt have some traits.

**Comment by willemolding** [comment]
Made some changes so that it will build properly with different feature flag configurations

**Pull request merged**